packages/wallet/primitives/src/network.ts

```diff
@@ -1,630 +1,1229 @@
-export type Network = {
+export enum NetworkType {
+  MAINNET = 'mainnet',
+  TESTNET = 'testnet',
+  LOCAL = 'local',
+}
+
+export type BlockExplorerConfig = {
+  name?: string
+  rootUrl: string
```

Comment:
should rename `rootUrl` to  `url` for simplicity. This will then also maps directly to the viem/wagmi BlockExplorer interface